File: cpp/3516_find_closest/find_closest.cpp
#include <cassert>
#include <cstdlib>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
 public:
  int findClosest(int x, int y, int z) {
    int dxz = abs(x - z), dyz = abs(y - z);
    if (dxz < dyz) {
      return 1;
    } else if (dxz > dyz) {
      return 2;
    } else {
      return 0;
    }
  }
};

int main() {
  vector<tuple<int, int, int, int>> tests{
      {2, 7, 4, 1},
      {2, 5, 6, 2},
  };

  for (auto &[x, y, z, expected] : tests) {
    assert(Solution().findClosest(x, y, z) == expected);
  }
}

File: rust/3516_find_closest/find_closest.rs
struct Solution;

impl Solution {
    pub fn find_closest(x: i32, y: i32, z: i32) -> i32 {
        let dxz = (x - z).abs();
        let dyz = (y - z).abs();
        if dxz < dyz {
            1
        } else if dxz > dyz {
            2
        } else {
            0
        }
    }
}

fn main() {
    let tests = vec![(2, 7, 4, 1), (2, 5, 6, 2)];

    for (x, y, z, expected) in tests {
        assert_eq!(Solution::find_closest(x, y, z), expected);
    }
}
